jerry-main/main-unix.c

```diff
@@ -423,182 +458,93 @@ main (int argc,
     }
     else if (!strcmp ("--mem-stats", argv[i]))
     {
-      if (jerry_is_feature_enabled (JERRY_FEATURE_MEM_STATS))
+      if (CHECK_FEATURE (JERRY_FEATURE_MEM_STATS, argv[i]))
       {
         jerry_port_default_set_log_level (JERRY_LOG_LEVEL_DEBUG);
         flags |= JERRY_INIT_MEM_STATS;
       }
-      else
-      {
-        jerry_port_default_set_log_level (JERRY_LOG_LEVEL_WARNING);
-        jerry_port_log (JERRY_LOG_LEVEL_WARNING,
-                        "Ignoring 'mem-stats' option because this feature is disabled!\n");
-      }
     }
     else if (!strcmp ("--mem-stats-separate", argv[i]))
     {
-      if (jerry_is_feature_enabled (JERRY_FEATURE_MEM_STATS))
+      if (CHECK_FEATURE (JERRY_FEATURE_MEM_STATS, argv[i]))
       {
         jerry_port_default_set_log_level (JERRY_LOG_LEVEL_DEBUG);
         flags |= JERRY_INIT_MEM_STATS_SEPARATE;
       }
-      else
-      {
-        jerry_port_default_set_log_level (JERRY_LOG_LEVEL_WARNING);
-        jerry_port_log (JERRY_LOG_LEVEL_WARNING,
-                        "Ignoring 'mem-stats-separate' option because this feature is disabled!\n");
-      }
     }
     else if (!strcmp ("--parse-only", argv[i]))
     {
       is_parse_only = true;
     }
     else if (!strcmp ("--show-opcodes", argv[i]))
     {
-      if (jerry_is_feature_enabled (JERRY_FEATURE_PARSER_DUMP))
+      if (CHECK_FEATURE (JERRY_FEATURE_PARSER_DUMP, argv[i]))
       {
         jerry_port_default_set_log_level (JERRY_LOG_LEVEL_DEBUG);
         flags |= JERRY_INIT_SHOW_OPCODES;
       }
-      else
-      {
-        jerry_port_default_set_log_level (JERRY_LOG_LEVEL_WARNING);
-        jerry_port_log (JERRY_LOG_LEVEL_WARNING,
-                        "Ignoring 'show-opcodes' option because this feature is disabled!\n");
-      }
     }
     else if (!strcmp ("--show-regexp-opcodes", argv[i]))
     {
-      if (jerry_is_feature_enabled (JERRY_FEATURE_PARSER_DUMP))
+      if (CHECK_FEATURE (JERRY_FEATURE_PARSER_DUMP, argv[i]))
```

Comment:
Question to self/anyone: was this OK before the patch? Shouldn't this line check for `JERRY_FEATURE_REGEXP_DUMP`?